frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -391,6 +393,37 @@ def circuit(x):
         _, observed_shape = tree_flatten(observed)
         assert expected_shape == observed_shape
 
+    def test_mcm_method_with_zne(self, backend):
+        """Test that the dynamic_one_shot works with ZNE."""
+        dev = qml.device(backend, wires=1, shots=5)
+
+        def circuit():
+            return qml.expval(qml.PauliZ(0))
+
+        mcm_method = "one-shot"
+
+        @qjit
+        def mitigated_circuit():
+            s = jnp.array([1, 2])
+            g = qml.QNode(circuit, dev, mcm_method=mcm_method)
+            return mitigate_with_zne(g, scale_factors=s)()
+
+        observed = mitigated_circuit()
+        mcm_method = None
+        expected = mitigated_circuit()
+
+        assert np.allclose(expected, observed)
```

Comment:
Aren't we testing it against itself here? 🤔

Reply:
I think python late binding means reference to globals within functions will actually track the latest value of a global symbol, although not entirely sure here. In any case I agree that the test should make two qnodes explicitly, one for each `mcm_method`, and compare them.

Reply:
Oh I see, yeah I missed that.

Reply:
But in this case we are not invoking the python function twice, only the compiled function, which is not going to see any changes in the `mcm_method` variable.

Reply:
Ah true, the second call doesn't go through python at all. Then it should be two qnodes.

Reply:
Ah good catch! I forgot I am calling compiled funnction :) Fixed it.